src/io/vtk_writer.cc

```diff
@@ -41,6 +41,51 @@ void VtkWriter::write_geometry(const std::string& filename) {
   writer->Write();
 }
 
+//! Write Tensor data
+void VtkWriter::write_tensor_point_data(
+    const std::string& filename,
+    const std::vector<Eigen::Matrix<double, 6, 1>>& data,
+    const std::string& data_field) {
+
+  // Create a polydata to store everything in it
+  auto pdata = vtkSmartPointer<vtkPolyData>::New();
+
+  // Add the points to the dataset
+  pdata->SetPoints(points_);
+
+  // Create an array to hold distance information
+  auto tensordata = vtkSmartPointer<vtkDoubleArray>::New();
+  tensordata->SetNumberOfComponents(9);
+  tensordata->SetNumberOfTuples(pdata->GetNumberOfPoints());
+  tensordata->SetName(data_field.c_str());
+
+  // Evaluate the signed distance function at all of the grid points
+  for (vtkIdType id = 0; id < pdata->GetNumberOfPoints(); ++id) {
+    const double* vdata = data.at(id).data();
+    tensordata->InsertTuple9(id, vdata[0], vdata[3], vdata[4], vdata[3],
```

Comment:
I think there is an error here. Here, you are putting vdata[4] in the x-z direction. However, in our code, vdata[5] should be in x-z direction as:
https://github.com/cb-geo/mpm/blob/72cd3842961f89962ac66b87b823c5674f9ce2f9/include/particles/particle.tcc#L546-L554

Reply:
Sorry my bad, I made a mistake when entering these values in the matrix